pkg/cmd/container/exec.go

```diff
@@ -133,13 +133,10 @@ func execActionWithContainer(ctx context.Context, client *containerd.Client, con
 		return nil
 	}
 	status := <-statusC
-	code, _, err := status.Result()
+	_, _, err = status.Result()
 	if err != nil {
 		return err
 	}
-	if code != 0 {
```

Comment:
@zouyee not displaying the message is one thing, but clearly we still need to error out.

Reply:
The executable we ran inside the container that errored, we need to print error ?

Reply:
We need to return an error from this method, otherwise nerdctl will exit 0.

Reply:
Why does exec function care about the results of execution inside the container?

Reply:
It should be just same as Docker

Reply:
nerdctl exec
```
nerdctl -n k8s.io exec  xxx bash -c  'bash ./faketime.sh xxx';echo $?
tomcat: stopped
tomcat: started
FATA[0005] exec failed with exit code 1
1
```

docker exec
```
docker exec xxx  bash -c  'bash ./faketime.sh xxx';echo $?
tomcat: stopped
tomcat: started
1
```

Reply:
Yes, we should do the same thing as docker.
With your patch right now, we no longer return exit 1 because you do not return a go error.

Reply:
Agree with you @zouyee  you may write stderr to `/dev/null` as workaround